components/telemetry-operator/internal/fluentbit/utils.go

```diff
@@ -0,0 +1,155 @@
+
+	if err := f.client.Patch(ctx, &patchedDS, client.MergeFrom(&ds)); err != nil {
+		log.Error(err, "Failed to patch Fluent Bit to trigger rolling update")
+		return err
+	}
+	restartsTotal.Inc()
+	return nil
+}
+
+func (f *DaemonSetUtils) IsFluentBitDaemonSetReady(ctx context.Context) (bool, error) {
+	log := logf.FromContext(ctx)
+	var ds appsv1.DaemonSet
+	if err := f.client.Get(ctx, f.FluentBitDaemonSet, &ds); err != nil {
+		log.Error(err, "Failed getting fluent bit daemon set")
+		return false, err
+	}
+
+	generation := ds.Generation
+	observedGeneration := ds.Status.ObservedGeneration
+	updated := ds.Status.UpdatedNumberScheduled
+	desired := ds.Status.DesiredNumberScheduled
+	ready := ds.Status.NumberReady
+
+	log.V(1).Info(fmt.Sprintf("Checking fluent bit: updated: %d, desired: %d, ready: %d, generation: %d, observed generation: %d",
+		updated, desired, ready, generation, observedGeneration))
+
+	return observedGeneration == generation && updated == desired && ready >= desired, nil
+}
+
+func (f *DaemonSetUtils) GetFluentBitConfig(ctx context.Context,
+	currentBaseDirectory, fluentBitParsersConfigMapKey string,
+	fluentBitConfigMap types.NamespacedName,
+	pipelineConfig PipelineConfig,
+	pipeline *telemetryv1alpha1.LogPipeline,
+	parser *telemetryv1alpha1.LogParser) ([]fs.File, error) {
+	var configFiles []fs.File
+	fluentBitSectionsConfigDirectory := currentBaseDirectory + "/dynamic"
+	fluentBitParsersConfigDirectory := currentBaseDirectory + "/dynamic-parsers"
+	fluentBitFilesDirectory := currentBaseDirectory + "/files"
+
+	var generalCm v1.ConfigMap
+	if err := f.client.Get(ctx, fluentBitConfigMap, &generalCm); err != nil {
+		return []fs.File{}, err
+	}
+	for key, data := range generalCm.Data {
+		configFiles = append(configFiles, fs.File{
+			Path: currentBaseDirectory,
+			Name: key,
+			Data: data,
+		})
+	}
+	var logPipelines telemetryv1alpha1.LogPipelineList
+	err := f.client.List(ctx, &logPipelines)
+	if err != nil {
+		return []fs.File{}, err
+	}
+	if pipeline != nil {
+		logPipelines.Items = append(logPipelines.Items, *pipeline)
+	}
+	// Build the config from all the exiting pipelines
+	for _, logPipeline := range logPipelines.Items {
+		configFiles, err = appendFBConfigFile(configFiles, logPipeline, pipelineConfig, fluentBitSectionsConfigDirectory, fluentBitFilesDirectory)
+		if err != nil {
+			return []fs.File{}, err
+		}
+	}
+
+	var parsersConfig string
+	var logParsers telemetryv1alpha1.LogParserList
+	if err := f.client.List(ctx, &logParsers); err != nil {
+		return []fs.File{}, err
+	}
+	if parser != nil {
+		logParsers.Items = append(logParsers.Items, *parser)
+	}
+
+	parsersConfig = MergeParsersConfig(&logParsers)
+	configFiles = append(configFiles, fs.File{
+		Path: fluentBitParsersConfigDirectory,
+		Name: fluentBitParsersConfigMapKey,
+		Data: parsersConfig,
+	})
+
+	return configFiles, nil
+}
+func appendFBConfigFile(
```

Comment:
Let's use `FluentBit` in the function name. The other function names also do not abbreviate `DaemonSet`.